PRI-503: Prepare for docker upgrade to docker-ce

fatal-exception opened the pull request:
Package name has changed from docker-engine to docker-ce. We need to
support both until the whole environment has been upgraded

fatal-exception committed:
PRI-503: Prepare for docker upgrade to docker-ce

Package name has changed from docker-engine to docker-ce. We need to
support both until the whole environment has been upgraded

wunzeco commented on defaults/main.yml:
```diff
@@ -8,10 +8,21 @@ docker_apt_repo: "deb https://apt.dockerproject.org/repo ubuntu-{{ ansible_distr
+docker_package: docker-engine
+docker_package_version:  '*'    # specify '*' for latest version
+
 docker_engine_version:  '*'    # specify '*' for latest version
```
Pls consider replacing `docker_engine_version` with the new `docker_package_version` since they both relate to `docker_package`

fatal-exception committed:
PRI-503: Simplify docker-engine logic

We now only use the docker_package_version variable, so can remove logic to
check if we are installing docker-engine or something else

fatal-exception commented:
This PR MUST be merged at same time as https://github.com/o2-priority/infra-priority-stack/pull/367
